shared/actions/__tests__/signup.test.js

```diff
@@ -0,0 +1,393 @@
+// @flow
+/* eslint-env jest */
+import * as Constants from '../../constants/signup'
+import * as SignupGen from '../signup-gen'
+import * as LoginGen from '../login-gen'
+import * as Saga from '../../util/saga'
+import type {TypedState} from '../../constants/reducer'
+import {loginTab} from '../../constants/tabs'
+import HiddenString from '../../util/hidden-string'
+import {navigateUp, navigateTo, navigateAppend} from '../route-tree'
+import {_testing} from '../signup'
+import reducer from '../../reducers/signup'
+
+jest.unmock('immutable')
+
+describe('resetNav works', () => {
+  it('nab based on login', () => {
+    expect(_testing.resetNav()).toEqual(Saga.put(LoginGen.createNavBasedOnLoginAndInitialState()))
+  })
+})
+
+describe('goBackAndClearErrors', () => {
+  it('errors get cleaned and we go back a level', () => {
+    const getState = () => ({
+      signup: Constants.makeState({
+        devicenameError: 'bad name',
+        emailError: 'bad email',
+        inviteCodeError: 'bad invite',
+        nameError: 'bad name',
+        passphraseError: new HiddenString('bad pass'),
+        signupError: new HiddenString('bad signup'),
+        usernameError: 'bad username',
+      }),
+    })
+
+    const action = SignupGen.createGoBackAndClearErrors()
+    const nextState = {signup: reducer(getState().signup, action)}
+    expect(nextState.signup.devicenameError).toEqual('')
```

Comment:
Would it be better to do something like: `expect(nextState.signup).toEqual(Constants.makeState({ /* expected signup state */}))`

Reply:
unclear, seems like you'd have to fill in a bunch more stuff